src/linker/Linker/MessageContainer.cs

```diff
@@ -256,10 +256,8 @@ public string ToMSBuildString ()
 			if (Origin?.Provider != null) {
 				if (Origin?.Provider is MethodDefinition method)
 					sb.Append (method.GetDisplayName ());
-				else if (Origin?.Provider is TypeDefinition type)
+				else if (Origin?.Provider is MemberReference type)
```

Comment:
While I think this is in general correct, it feels dangerous. If there can be an instance of `IMemberDefinition` which is not `MemberReference` we would now throw `NotSupportedException`. Which would be a regression. Quickly looking through Cecil, this should not be the case, but I don't know for sure.
Would be safer to keep the `IMemberDefinition` case as a fallback.